src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -766,17 +773,32 @@ private async Task<(int totalCount, List<string> resourcesTypes)> CalculateTotal
             return (totalCount, resourcesTypes);
         }
 
-        private async Task UpdateParametersAndCompleteJob()
+        private async Task UpdateParametersAndCompleteJob(CancellationToken cancellationToken)
         {
             // here we check if all the resource types which are base types of the search parameter
             // were reindexed by this job.  If so, then we should mark the search parameters
             // as fully reindexed
             var fullyIndexedParamUris = new List<string>();
             var reindexedResourcesSet = new HashSet<string>(_reindexJobRecord.Resources);
+            var searchParamStatusCollection = await _searchParameterStatusManager.GetAllSearchParameterStatus(cancellationToken);
```

Comment:
Seeing this request to update the current status reminds me that perhaps we need to block any requests to change SearchParam status while a reindex job is running.

We have this check when a user creates or updates a CustomSearch parameter, as we don't want them introducing problems for the currently running reindex job.  We should think about adding this same restriction for the $status operation.  This is currently done for Search Parameters in this file: \fhir-server\src\Microsoft.Health.Fhir.Shared.Core\Features\Search\Parameters\SearchParameterValidator.cs.

This is not something that needs to be done in this PR, but we should capture the need to analyze this.

Reply:
That is a good point. I will make note of it.